### ISSUE TITLE
Sort by recently used Windows 

### ISSUE THREAD

**Issue by aloispichler** [H3rmt/hyprswitch#5]
Instead of sorting, is it possible to toggle along the history of activated windows (back, and forward); that is, in most-recently-used order?
This is actually the default behavior in most other alttabs (cf. https://github.com/sagb/alttab).

**Comment by H3rmt** [comment]
I'm not entirely sure if Hyprland has the capability to return a list of recently visited windows, i'll get back to you once I've gathered more information on this.

**Comment by aloispichler** [comment]
Thank you. If this information is not provided by hyprland itself, then it would be a good escape to just use the windowswitcher’s most-recently-used order.

**Comment by aloispichler** [comment]
I would desperately need this functionality.

**Comment by aloispichler** [comment]
The last property in `hyprctrl clients` is `focusHistoryID` and it does the job. I hope this will help!

**Comment by H3rmt** [comment]
the problem with implementing this feature is that hyprland-rs (the library used to interface with hyprland) currently doesn't focusHistoryID.
There is a commit

**Comment by aloispichler** [comment]
Seems that you are working on this option, which I really appreciate. Thank you.

The role model is of course the option ’Ctrl+Tab cycles through tabs in recently used order’ in firefox,
which enables ‘Ctrl+Tab’ and ‘Ctrl+Shift+Tab’.

**Comment by H3rmt** [comment]
sorting by recently used windows is now supported, but using it will just cycle through the current and last used window, as the order of focused windows changes on first execution

**Comment by H3rmt** [comment]
@aloispichler there might be the a better solution available when running in daemon-mode.
The Daemon could load the order in which windows were selected on start, so you could switch to e.g. the 3. most recently used window.
I don't see a possibility for implementing this in CLI mode because the script wouldn't know whether to use the previous or the current order.
However, you could use the --offset option to directly switch to the 3rd element, but this doesn't seem practical to me.

(the --offset option and daemon/gui mode will be available in the next release)

**Comment by aloispichler** [comment]
Seems to be some excellent work which I will try soon. Look forward to seeing the Arch package!

**Comment by H3rmt** [comment]
added in latest release

**Comment by aloispichler** [comment]
I was working with your app today. Nice experience, indeed! Thank you. I used the AUR-version.
Here are few points:
* If `--sort-recent` is selected, then the GUI should `--sort-recent` as well.
* Cycling should NOT reveal the window: the window should reveal only after being finally selected.
* It would be desirable to select the next window by `$mod+$key+…+$key$` at key release. Now, this is `$mod+key$`, …, `$mod+key$` and `$mod$` (extra release) to stop the deamon.

All three points above are the default behavior of `Ctrl-Tab` in Firefox (with setting 'Ctrl+Tab cycles through tabs in recently used order').
Overall: very pleasant experience, thanks again!

**Comment by H3rmt** [comment]
> It would be desirable to select the next window by $mod+$key+…+$key$ at key release. Now, this is $mod+key$, …, $mod+key$ and $mod$ (extra release) to stop the deamon.

If i understand this correctly you currently use the GUI Config.
Repeated switching without releasing the $mod key / holding $key is possible with the GUI + Keyboard Config. (This however stops the daemon after releasing the $mod key, so you $mod+$key+…+$key-$mod to switch and close) (This currently switches at every command and not after the daemon exits)

**Comment by H3rmt** [comment]
Could you provide the part of your hyprland config that calls hyprswitch.
This would allow me to debug the First Problem you mentioned

**Comment by aloispichler** [comment]
I use:
```
# open hyprswitch
$key = TAB
$modifier = SUPER
$switch_release = SUPER_L
$reverse = SHIFT

# open hyprswitch
bind=$modifier, $key, exec, hyprswitch --ignore-workspaces --sort-recent --do-initial-execute --daemon
bind=$modifier $reverse, $key, exec, hyprswitch --ignore-workspaces --sort-recent --do-initial-execute --daemon -r

# close hyprswitch
bindr=$modifier, $switch_release, exec, hyprswitch --stop-daemon
```

**Comment by H3rmt** [comment]
This config is intended to be used if you want to click on the windows in the GUI
If you want to want a config with more default keyboard shortcuts you could try the GUI + Keyboard Config.
If you just want to be able to hold down the $key key to continue switch windows without pressing the $modifier key every time you should use ``binde=$modifier, $key, ...`` (the `e` repeats the command)

**Comment by aloispichler** [comment]
Aaaah, yes indeed, amazing! I am sorry I have misunderstood the usage in the read.me.
I really like this app.

**Comment by H3rmt** [comment]
> If --sort-recent is selected, then the GUI should --sort-recent as well.

Could you try to try to give more detailed explanation on what you did, what happened and what you expected to happen (maybe in a new issue?)
